source/simulator/lateral_averaging.cc

```diff
@@ -725,6 +761,14 @@ namespace aspect
                                (this->introspection().extractors.velocities,
                                 &this->get_gravity_model()));
           }
+        else if (property_names[property_index].substr(property_names[property_index].size()-5) == "_mass")
```

Comment:
I am slightly concerned this creates conflicts with any other property called `something_mass`. Could you make this more secure by checking that the first part fits a compositional field and the last part is `_mass`? E.g.
```
else if (this->introspection().compositional_name_exists(property_names[property_index].substr(0, property_names[property_index].size()-5)) &&
property_names[property_index].substr(property_names[property_index].size()-5) == "_mass")
```